### PR TITLE
Is16 create menu class

### PR DESCRIPTION
**Pull request by Arian1192**
The implementation of the main menu is created to choose between data entry and school management, unit tests are made to discern between positive and negative cases in reference to the possible related casuistry.

- [x] tested

### CODE REVIEW

File: src/main/java/Menu.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class Menu {
+    private static final int OPTION_DATA_ENTRY = 1;
+    private static final int OPTION_SCHOOL_MANAGEMENT = 2;
+
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        String welcomeMenu = FigletFont.convertOneLine("Iron-School");
```
Comment (sergi-svg) [inline src/main/java/Menu.java]: I'd rather use this approach, so we can avoid the creation of a just single use variable.
System.out.println(FigletFont.convertOneLine("Iron-School"));

File: src/main/java/Menu.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        System.out.println(welcomeMenu);
+        int selectedOption = showPrincipalMenuAndRetrieveOption(sc);
+        System.out.println("Option Selected: " + selectedOption);
+    }
+
+        public static int showPrincipalMenuAndRetrieveOption(Scanner sc) {
+            System.out.println("Select option");
+            System.out.println(OPTION_DATA_ENTRY + " -> Data Entry");
```
Comment (sergi-svg) [inline src/main/java/Menu.java]: Here perhaps is better to use an Enum and print its values. Somthing like..
Stream.of(MainMenu.values())
                        .forEach(entry -> System.out.println((entry.ordinal() + 1) + " - " + entry.getDescription()));

And the Enum could be..
public enum MainMenu {
    OPTION_DATA_ENTRY("Data Entry"),
    OPTION_SCHOOL_MANAGEMENT("School Management");

    private final String description;

    MainMenu(String description) {
        this.description = description;
    }

    public String getDescription() {
        return description;
    }
}

**Comment by Arian1192** [comment]
I have to polish the tests before it merge
*EDIT

- [x] tested

File: src/main/java/Menu.java (hunk 1 above)
Comment (Arian1192) [reply src/main/java/Menu.java]: refactored thx

File: src/main/java/Menu.java (hunk 2 above)
Comment (Arian1192) [reply src/main/java/Menu.java]: Your advice has been implemented and refactored and test refactored too

**Comment by Arian1192** [comment]
Captura de pantalla 2024-02-27 a las 16 01 02

I add the test coverage

File: src/main/java/MainMenuOptions.java
Comment (sergi-svg) [inline src/main/java/MainMenuOptions.java:1]: The names of enum are usually in singular, so it should be MainMenuOption.java or  MainMenu.java
And we could locate all enums in the utils package

File: src/main/java/MainMenuOptions.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import lombok.Getter;
+
+@Getter
```
Comment (sergi-svg) [inline src/main/java/MainMenuOptions.java]: First time I see lombok in aa enum, very good idea!

File: src/main/java/MainMenuOptions.java
Comment (Arian1192) [reply src/main/java/MainMenuOptions.java:1]: okei I'm going to move the enum to utils

**Review by sergi-svg — APPROVED** [review]

**Pull request merged**